src/Text/Pandoc/Builder.hs

```diff
@@ -566,6 +569,28 @@ simpleCaption = caption Nothing
 emptyCaption :: Caption
 emptyCaption = simpleCaption mempty
 
+-- | Bidirectional patter synonym
+--
+-- It can pass as a Block constructor
+--
+-- λ> SimpleFigure nullAttr [] (T.pack "") (T.pack "")
```

Comment:
Probably good to use the [Haddock examples syntax](https://haskell-haddock.readthedocs.io/en/latest/markup.html#examples) for this.